client/src/components/Tool/ToolCard.vue

```diff
@@ -1,295 +1,166 @@
-                }
-            );
-        },
-        onCopyLink() {
-            copyLink(this.id, "Link was copied to your clipboard");
-        },
-        onCopyId() {
-            copyId(this.options.id, "Tool ID was copied to your clipboard");
-        },
-        onDownload() {
-            downloadTool(this.id);
-        },
-        onLink() {
-            openLink(this.options.sharable_url);
-        },
-        getFavorites() {
-            const preferences = this.user.preferences;
-            if (preferences && preferences.favorites) {
-                return JSON.parse(preferences.favorites);
-            } else {
-                return {
-                    tools: [],
-                };
-            }
-        },
-        updateFavorites(objectType, newFavorites) {
-            const favorites = this.getFavorites();
-            favorites[objectType] = newFavorites[objectType];
-            this.$emit("onUpdateFavorites", this.user, JSON.stringify(favorites));
-        },
+    disabled: {
+        type: Boolean,
+        default: false,
     },
-};
+});
+
+const emit = defineEmits(["onChangeVersion"]);
+
+function onChangeVersion(v) {
+    emit("onChangeVersion", v);
+}
+
+const errorText = ref(null);
+
+watch(
+    () => props.id,
+    () => {
+        errorText.value = null;
+    }
+);
+
+function onSetError(e) {
+    errorText.value = e;
+}
+
+const { currentUser: user } = useCurrentUser();
+const hasUser = computed(() => !user.value.isAnonymous);
+
+const versions = computed(() => props.options.versions);
+const showVersions = computed(() => props.options.versions?.length > 1);
 </script>
-<style scoped>
+
+<template>
+    <div class="position-relative">
+        <div class="underlay sticky-top" />
+        <div class="tool-header sticky-top bg-secondary px-2 py-1 rounded">
+            <div class="d-flex justify-content-between">
+                <div class="py-1 d-flex flex-wrap flex-gapx-1">
+                    <span>
+                        <icon icon="wrench" class="fa-fw" />
+                        <h1 itemprop="name">{{ title }}</h1>
+                    </span>
+                    <span itemprop="description">{{ description }}</span>
+                    <span>(Galaxy Version {{ version }})</span>
+                </div>
+                <div class="d-flex flex-nowrap align-items-start">
+                    <b-button-group class="tool-card-buttons">
+                        <ToolFavoriteButton v-if="hasUser" :id="props.id" @onSetError="onSetError" />
+                        <ToolVersionsButton
+                            v-if="showVersions"
+                            :version="props.version"
+                            :versions="versions"
+                            @onChangeVersion="onChangeVersion" />
+                        <ToolOptionsButton :id="props.id" :sharableUrl="props.options.sharable_url" />
+                    </b-button-group>
+                    <slot name="header-buttons" />
```

Comment:
Might require some padding, or the header-buttons should be integrated into the button-group: 
<img width="158" alt="image" src="https://user-images.githubusercontent.com/2105447/187920772-8de72d27-97de-4495-b2a0-c1804fef3af2.png">